### DISCUSSION TITLE
NAS folder not mounted after reboot & MPD stops running when external DAC switched off

### DISCUSSION THREAD

**Discussion by erivet1** [rern/rAudio#230] [Bug Reports]
Hi,

I am using the rAudio 1 version 20210227 on a RPi3 model B. After several hours looking at various threads in runeaudio.com, I have never been able to mount any NAS folder via CIFS locally (although it works on my mac). However I am able to mount NAS folders via nfs but only manually through the /etc/fstab/ file (it does not work via the UI). By the way in Sources in UI, I have only _Devices_ and _File System Table_, but not _Mounts_ (as it appears to be in the _User Guide_). After rebooting RPi I have a red dot on my /mnt/MPD/NAS/xxx. To mount it I need to do 'sudo mount -a' in the terminal via SSH or a 'remount' in the UI. I tried different options in /etc/fstab/ such as defaults,auto,_netdev.

Does someone have some suggestions? 
Thank you and well done for this nice work.

**Comment by firstchanceongit** [comment]
It works fine on my installation (a setup with a Raspberry based NAS with the music collection and a separate Raspberry with the raudio). Following startup of the NAS it needs, however, a little while (probably a minute or so) until the NAS' network address is "known" in the network. If I observe such startup waiting period, the raudio Raspberry has no problem connecting it at startup. But I thought about a suggestion to rern whether a regular "mount -a" command (like every minute or so) or a tleast a one-time repeat of a "mount -a" about a minute after startup of raudio would make sense.)

By the way (also suggestion to rern) - it would be great if raudio would not only allow for static IP for the NAS in the UI but also for host names (including such from a Windows network). Currently, entering host names is not allowed/possible in the UI.

**Reply by erivet1** [reply]
I entered a static IP to mount the NAS folder, and my (Syno) NAS has not been powered down during my tries. Do I need to remount it manually after startup? I am surprised I have such issues with my setup. I gave a Read only privilege, map all users to admin, and enabled the 3 other check boxes in Syno for the specific local IPs (wifi & ethernet) for RPi. 

Sometimes the MPD stops running.

**Reply by firstchanceongit** [reply]
Do you connect via ethernet or wifi? I seem to recall that experienced similar issues when I was connecting via an (unstable) wifi connection.

**Reply by erivet1** [reply]
The RPi is connected via wifi (around 4m from router).

**Reply by erivet1** [reply]
As said above, sometimes the MPD stops running. With a _systemctl status mpd_ command, I get this:

* mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; vendor preset: disabled)
    Drop-In: /etc/systemd/system/mpd.service.d
             `-override.conf
     Active: failed (Result: timeout) since Sun 2021-02-28 15:08:17 CET; 2min 40s ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
    Process: 9004 ExecStart=/usr/bin/taskset -c 3 /usr/bin/mpd --no-daemon (code=killed, signal=KILL)
   Main PID: 9004 (code=killed, signal=KILL)

Feb 28 15:05:17 rAudio systemd[1]: Starting Music Player Daemon...
Feb 28 15:05:17 rAudio taskset[9004]: exception: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
Feb 28 15:06:47 rAudio systemd[1]: mpd.service: start operation timed out. Terminating.
Feb 28 15:08:17 rAudio systemd[1]: mpd.service: State 'stop-sigterm' timed out. Killing.
Feb 28 15:08:17 rAudio systemd[1]: mpd.service: Killing process 9004 (mpd) with signal SIGKILL.
Feb 28 15:08:17 rAudio systemd[1]: mpd.service: Main process exited, code=killed, status=9/KILL
Feb 28 15:08:17 rAudio systemd[1]: mpd.service: Failed with result 'timeout'.
Feb 28 15:08:17 rAudio systemd[1]: Failed to start Music Player Daemon.

And with the _netstat -anp | grep 6600_ command, I get this:
tcp        1      0 127.0.0.1:51344         127.0.0.1:6600          CLOSE_WAIT  331/upmpdcli

**Reply by yopmismo** [reply]
Today I made a clean install in my RPI 4 8GB and after hours trying to figure the problems I give up.
USB RAID is mounted, but library = 0.
MPD not running:

# pacman -Q mpd; systemctl status mpd

mpd 0.22.6-1
● mpd.service - Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; vendor preset: disabled)
    Drop-In: /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: failed (Result: exit-code) since Mon 2021-03-01 17:12:17 -03; 6min ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
    Process: 447 ExecStart=/usr/bin/taskset -c 3 /usr/bin/mpd --no-daemon (code=exited, status=1/FAILURE)
   Main PID: 447 (code=exited, status=1/FAILURE)



What can I do?

At this time I'm running without problems original Runeaudio in RPI 3 1GB.

**Reply by rern** [reply]
@erivet1,
Try:
 `sed -i 's/1..10/1..30/' /srv/http/bash/startup.sh`
(extend checking for NAS ready to 30 seconds)
Then reboot.

@yopmismo
Try:
- Disconnect USB drive
- Connect a USB thumb drive with a couple of albums
- Connect the USB RAID (wait for Library update to finish)

**Reply by yopmismo** [reply]
Thank you @rern
It worked, I never figured this solution...

**Reply by erivet1** [reply]
Hi @rern,

Thanks for answering, I greatly appreciate. I flashed the SD and started with a clean install. 
I sticked with ethernet connection, mounted the NAS music folder, and added your command. 
I do see the change in the startup script.
Unfortunately I have to mount the folder manually after startup. 
Any other suggestions?

**Comment by erivet1** [comment]
Actually I found out when the MPD stops working. It is when I unplugged the USB cable on my USB DAC or I switch off the DAC. 
I disabled the headphone output in boot/config.txt (dtparam=audio=off) but I still have this problem. It means that every time I switch on or off the external DAC, I need to reboot the RPi. 
@rern I saw that you have a similar config with an eternal SMSL DAC, did you experience such bug?

**Reply by rern** [reply]
Now MPD runs with a single audio output at a time. No needs to disable the on-board one.

By default:
- I2S DACs - On-board disabled when setup.
- USB DACs - On-board still enabled. There should be at least 1 output available, when power off, for other functions and features to work properly.

**Reply by erivet1** [reply]
But even if I keep the headphone output enabled, the MPD stops working when I turn off the USB DAC. I do not think this is normal, correct?

**Reply by rern** [reply]
My USB DAC has never behaved the way yours did. Was it powered separately or by the USB port? 

My RPi and DAC are powered by a linear power supply. It logged "under-voltage" event when rebuilt new database from my USB harddrive, powered by the USB port itself. Sometimes MPD died half way through.

See the System page if there is a "Voltage under 4.7V" line in Status section when MPD died.
